src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -329,91 +337,103 @@ uct_ib_mlx5_set_ctrl_qpn_ds(struct mlx5_wqe_ctrl_seg *ctrl, uint32_t qp_num,
 static UCS_F_ALWAYS_INLINE void
 uct_ib_mlx5_set_ctrl_seg(struct mlx5_wqe_ctrl_seg* ctrl, uint16_t pi,
                          uint8_t opcode, uint8_t opmod, uint32_t qp_num,
-                         uint8_t fm_ce_se, unsigned wqe_size)
+                         uint8_t fm_ce_se, unsigned wqe_size,
+                         uint16_t dci_channel)
 {
     uint8_t ds = ucs_div_round_up(wqe_size, UCT_IB_MLX5_WQE_SEG_SIZE);
 #if defined(__ARM_NEON)
     uint8x16_t table = {1,               /* opmod */
                         5,  4,           /* sw_pi in BE */
                         2,               /* opcode */
                         14, 13, 12,      /* QP num */
-                        8,               /* data size */
+                        4,               /* data size */
                         16,              /* signature (set 0) */
-                        16, 16,          /* reserved (set 0) */
+                        9, 8,            /* dci stream channel ID in BE */
                         0,               /* signal/fence_mode */
                         16, 16, 16, 16}; /* immediate (set to 0)*/
-    uint32x4_t data = {(opcode << 16) | (opmod << 8) | (uint32_t)fm_ce_se,
-                       pi, ds, qp_num};
+    uint32x4_t data = {(ds << 24) | (opcode << 16) | (opmod << 8) | fm_ce_se,
+                       pi, dci_channel, qp_num};
+#elif !defined (__SSE4_2__)
+    uct_ib_mlx5_wqe_ctrl_seg_t *uct_ctrl = (uct_ib_mlx5_wqe_ctrl_seg_t*)ctrl;
```

Comment:
we should use uct_ib_mlx5_wqe_ctrl_seg_t instead of mlx5_wqe_ctrl_seg everywhere; not just cast to it here

Reply:
ok